### PR TITLE
MueLu: Replicate Prolongators for GMHD

### PR DESCRIPTION
**Pull request by rstumin**
Auto-PR for SHA 86c0319

### CODE REVIEW

File: packages/muelu/src/Transfers/Generic/MueLu_ReplicatePFactory_def.hpp [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  template <class Scalar,class LocalOrdinal, class GlobalOrdinal, class Node>
+  void ReplicatePFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::BuildP(Level& fineLevel,
+                                Level& coarseLevel) const {
+    FactoryMonitor m(*this, "Build", coarseLevel);
+
+    RCP<Matrix>      Psubblock             = coarseLevel.Get< RCP<Matrix> >("Psubblock", NoFactory::get());
+    const ParameterList& pL = GetParameterList();
+printf("before p print\n");
+std::cout << pL << std::endl;
+printf("after p print\n");
```
Comment (cgcgcg) [inline packages/muelu/src/Transfers/Generic/MueLu_ReplicatePFactory_def.hpp:104]: Remove?

File: packages/muelu/src/Transfers/Generic/MueLu_ReplicatePFactory_def.hpp [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    const LO dofPerNode = as<LO>(pL.get<int>   ("replicate: npdes"));
+
+printf("dofPerNode is %d\n",(int) dofPerNode);
```
Comment (cgcgcg) [inline packages/muelu/src/Transfers/Generic/MueLu_ReplicatePFactory_def.hpp:107]: Ditto

File: packages/muelu/src/MueCentral/MueLu_MasterList.cpp [hunk 3]
```diff
@@ -119,53 +119,54 @@ namespace MueLu {
...
+    if (name == "refmaxwell: use as preconditioner") { ss << "<Parameter name=\"refmaxwell: use as preconditioner\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
+    if (name == "refmaxwell: dump matrices") { ss << "<Parameter name=\"refmaxwell: dump matrices\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
+    if (name == "refmaxwell: subsolves on subcommunicators") { ss << "<Parameter name=\"refmaxwell: subsolves on subcommunicators\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
+    if (name == "refmaxwell: enable reuse") { ss << "<Parameter name=\"refmaxwell: enable reuse\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
+    if (name == "refmaxwell: skip first (1,1) level") { ss << "<Parameter name=\"refmaxwell: skip first (1,1) level\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
+    if (name == "refmaxwell: normalize nullspace") { ss << "<Parameter name=\"refmaxwell: normalize nullspace\" type=\"bool\" value=" << value << "/>"; return ss.str(); }
```
Comment (GrahamBenHarper) [inline packages/muelu/src/MueCentral/MueLu_MasterList.cpp:169]: It looks like the majority of this added whitespace at the end of every line? it's not necessarily an issue with the PR, but I might look at the scripts and see why that happens...

**Review by GrahamBenHarper — APPROVED** [review]
Looks good to me! There's one comment, but no need to address it if everything passes the AT.

File: packages/muelu/src/Transfers/Generic/MueLu_ReplicatePFactory_def.hpp (hunk 1 above)
Comment (rstumin) [reply packages/muelu/src/Transfers/Generic/MueLu_ReplicatePFactory_def.hpp]: Oops ... I missed this one and the following one.  Just fixed it. Thanks for catching.

File: packages/muelu/src/MueCentral/MueLu_MasterList.cpp (hunk 3 above)
Comment (rstumin) [reply packages/muelu/src/MueCentral/MueLu_MasterList.cpp:169]: Yeah I don't know what causes this. I don't think it is a problem.

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**